packages/cli/lib/validators/marketplaceValidators/theme/SectionValidator.js

```diff
@@ -0,0 +1,116 @@
+} = require('@hubspot/cli-lib/templates');
+const BaseValidator = require('../BaseValidator');
+const { VALIDATOR_KEYS } = require('../../constants');
+
+const SECTION_LIMIT = 50;
+
+class SectionValidator extends BaseValidator {
+  constructor(options) {
+    super(options);
+
+    this.errors = {
+      LIMIT_EXCEEDED: {
+        key: 'limitExceeded',
+        getCopy: ({ limit, total }) =>
+          `Section limit exceeded. Themes can only have ${limit} sections, but this theme has ${total}`,
+      },
+      MISSING_LABEL: {
+        key: 'missingLabel',
+        getCopy: ({ filePath }) =>
+          `Missing required property for ${filePath}. The section is missing the "label" property`,
+      },
+      MISSING_SCREENSHOT_PATH: {
+        key: 'missingScreenshotPath',
+        getCopy: ({ filePath }) =>
+          `Missing required property for ${filePath}. The section is missing the "screenshotPath" property`,
+      },
+      MISSING_DESCRIPTION: {
+        key: 'missingDescription',
+        getCopy: ({ filePath }) =>
+          `Missing required property for ${filePath}. The section is missing the "description" property`,
+      },
+    };
+  }
+
+  // Validates:
+  // - All sections have a 'description' annotation
+  // - All sections have a "label" annotation
+  // - All sections have a "screenshotPath" annotation
+  // - Theme does not have more than SECTION_LIMIT sections
+
+  validate(files) {
+    let validationErrors = [];
+    let sectionCount = 0;
+
+    files.forEach(file => {
+      if (file) {
+        const annotations = getFileAnnotations(file);
+        const templateType = getAnnotationValue(
+          annotations,
+          ANNOTATION_KEYS.templateType
+        );
+        const isAvailableForNewContent = getAnnotationValue(
+          annotations,
+          ANNOTATION_KEYS.isAvailableForNewContent
+        );
+
+        if (templateType !== 'section') {
+          return;
+        }
+        sectionCount++;
+
+        if (isAvailableForNewContent !== 'false') {
+          const description = getAnnotationValue(
+            annotations,
+            ANNOTATION_KEYS.descirption
+          );
+          const label = getAnnotationValue(annotations, ANNOTATION_KEYS.label);
+          const screenshotPath = getAnnotationValue(
+            annotations,
+            ANNOTATION_KEYS.screenshotPath
+          );
+
+          if (!description) {
+            validationErrors.push(
+              this.getError(this.errors.MISSING_DESCRIPTION, file)
+            );
+          }
+          if (!label) {
+            validationErrors.push(
+              this.getError(this.errors.MISSING_LABEL, file)
+            );
+          }
+          if (!screenshotPath) {
+            validationErrors.push(
+              this.getError(this.errors.MISSING_SCREENSHOT_PATH, file)
+            );
+          }
+        }
+      } else {
+        validationErrors.push(
+          this.getError(this.errors.MISSING_TEMPLATE_TYPE, file)
```

Comment:
`MISSING_TEMPLATE_TYPE` doesn't appear to be defined in the errors object.

Reply:
This else check isn't needed since we will abort [here](https://github.com/HubSpot/hubspot-cli/pull/546/files#diff-547055959147bfda4ac86ee037e1eeb2d499ec437041df4b9196d541f799fd7eR61) if the template type is undefined. Also, [this else check](https://github.com/HubSpot/hubspot-cli/blob/ee1e9e802d1a8423d88adab3c31b8f56c4ad2bbe/packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js#L154-L158) in the template validator should already catch this case where the section doesn't have a template type type.

Reply:
Thanks, I removed the else check.